### PR TITLE
Allow changing pkv precision

### PR DESCRIPTION
**Pull request by AlexKoff88**
This PR is a revised version of https://github.com/huggingface/optimum-intel/pull/370.

### CODE REVIEW

**Comment by AlexKoff88** [comment]
@luo-cheng2021, please take a look and test this PR with BF16.

File: optimum/intel/openvino/modeling_decoder.py [hunk 1]
```diff
@@ -135,6 +144,48 @@ def __init__(
...
+        if not force_fp32:
+            device = self._device.upper()
+            inference_precision_hint = core.get_property(device, "INFERENCE_PRECISION_HINT")
+            # ov_config["INFERENCE_PRECISION_HINT"] may override the prefer precision
+            if self.ov_config:
+                inference_precision_hint = self.ov_config.get("INFERENCE_PRECISION_HINT", "")
+                if inference_precision_hint in STR_TO_OV_TYPE:
+                    pkv_precision = STR_TO_OV_TYPE[inference_precision_hint]
+
+        self._pkv_precision = Type.f32
```
Comment (echarlaix) [inline optimum/intel/openvino/modeling_decoder.py:161]: Can we remove ?
```suggestion
```

**Review by echarlaix — APPROVED** [review]
LGTM, left a couple of minor comments, thanks a lot @AlexKoff88

File: optimum/intel/openvino/modeling_decoder.py [hunk 2]
```diff
@@ -108,6 +110,9 @@ def __init__(
                 "`dynamic_shapes` was set to `False` but static shapes are not supported for causal language model. Please set `dynamic_shapes=True`."
             )
 
+        enable_compilation = kwargs.get("compile", True)
+        kwargs["compile"] = False  # avoid extra compilation in the base class
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply optimum/intel/openvino/modeling_decoder.py:114]: done

File: optimum/intel/openvino/modeling_decoder.py [hunk 3]
```diff
@@ -135,6 +144,48 @@ def __init__(
...
             )
 
+    def update_pkv_precision(self, force_fp32=False):
+        if not self.use_cache:
+            return
+
+        pkv_precision = Type.f32
+        if not force_fp32:
+            device = self._device.upper()
+            inference_precision_hint = core.get_property(device, "INFERENCE_PRECISION_HINT")
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply optimum/intel/openvino/modeling_decoder.py:157]: Done

File: optimum/intel/openvino/modeling_decoder.py (hunk 1 above)
Comment (AlexKoff88) [reply optimum/intel/openvino/modeling_decoder.py]: Fixed

**Comment by AlexKoff88** [comment]
It seems like CI is broken by SD update.

**Comment by luo-cheng2021** [comment]
> @luo-cheng2021, please take a look and test this PR with BF16.

BF16 case works as expected.

**Pull request merged**